Automation Examples/010.md

```diff
@@ -0,0 +1,200 @@
+# Process
+
+Open http://nseindia.com/corporates/content/eq_research_reports_listed.htm and for the first 50 companies, download a recent month’s report. Save it in a separate folder **Today’s date/Company Name/PDF Name.pdf**
+
+# Steps to automate
+
+#### 1. Create variables - `url` storing the link given and `basePath` containing the path to the main folder
+
+```G1ANT
+♥url = http://nseindia.com/corporates/content/eq_research_reports_listed.htm
+♥basePath = C:\Users\dell\Desktop\PDF Automation
+```
+
+#### 2. Create a main procedure in which you open the website and call it after the initialised variables section
+
+```G1ANT
+call DownloadPdfFiles
+
+procedure DownloadPdfFiles
+    selenium.open chrome url ♥url
+end procedure
+```
+
+#### 3. Run the script and investigate the website yourself, open developer tools and find there the elements that are important (company names and the buttons to open their webpages)
+
+After a quick investigation you should see that the company names are in the third column and you can get the first records by running these scripts in the developer tools console:
+
+```js
+document.getElementsByClassName("normalText")[2].innerText
+document.getElementsByClassName("normalText")[6].innerText
+document.getElementsByClassName("normalText")[10].innerText
+```
+Getting the buttons to open company webpages is quite similar. Here are examples of how you can open the first three of them:
+
+```js
+document.getElementsByClassName("normalText")[3].getElementsByTagName("a")[0].click()
+document.getElementsByClassName("normalText")[7].getElementsByTagName("a")[0].click()
+document.getElementsByClassName("normalText")[11].getElementsByTagName("a")[0].click()
+```
+
+You can notice that the index for each record increases by 4 and that the respective buttons' indexes are greater by 1 than the company names' indexes.
+
+#### 4. Based on your research, create a for loop in which you will wait for the elements to be loaded, get a company name and then open its webpage. 
+
+```G1ANT
+    for ♥i from 2 to 198 step 4
+        selenium.waitforvalue ‴return document.getElementsByClassName("normalText").length > 0‴ expectedvalue true
+        selenium.runscript ‴return document.getElementsByClassName("normalText")[♥i].innerText‴
+        ♥companyName = ♥result
+        selenium.runscript ‴document.getElementsByClassName("normalText")[♥i+1].getElementsByTagName("a")[0].click()‴
```

Comment:
How about using selenium.gettext instead of selenium.runscript? I think its way more user friendly and its purpose exactly what you are doing here - gettintext from element.

Reply:
@lfronczyk what do you think? Do I have to rewrite the whole tutorial and use more selenium commands? :)